### DISCUSSION TITLE
Does burn have a converter from PTH file

### DISCUSSION THREAD

**Discussion by longzou** [tracel-ai/burn#1465] [Q&A] [answered]
Mostly, there are many models saved by pytorch. So, does the burn have the converter to help this?

**Comment by laggui** [comment, marked as answer]
Yes! We have `PyTorchFileRecorder` (see [reference in the book](https://burn.dev/book/import/pytorch-model.html)).

Since PyTorch saves a (flattened) state dictionary without any model structure definition, you need to define the model in Burn to import the weights you want to reuse.

*Shameless plug:* if you're looking to port your model over to Burn, take a look at [this tutorial](https://dev.to/laggui/transitioning-from-pytorch-to-burn-45m) to port the `torchvision` ResNet weights to the Burn implementation.

If you don't want to redefine the model in Burn, you can also try to [import the ONNX model](https://burn.dev/book/import/onnx-model.html) directly. Since ONNX captures the weights and operations graph, we can automatically generate the module definition.

**Comment by antimora** [comment]
Also, on the main currently made my improvements. Here are the summary what's on the main but not in the released version:

Notable enhancements to importing PyTorch weights and configs:

1. Ability to extract a set of tensor weights associated with a map key.

2. Ability to extract a model configuration from a .pt file.

3. Print tensor keys with shape/type info when a debug flag is enabled. You can also see new/old remapped keys.

4. Enum module support. Weights are automatically imported. You can define module variants, and the import process will pick the correct Enum.

5. Support for non-contiguous indexes in the source file, such as 0, 2, 4, 6. PyTorch skips exporting modules without weights, which is why you may have gaps in the indexes.

6. Improved error messages.

7. PyTorchFileRecorder now works on Windows.

8. Other PyTorch .pt format fixes (Fortran-contiguous layout, support for newer PyTorch formats).

The book is updated which is on the main too and currently not published.
